datafusion/expr/src/signature.rs

```diff
@@ -95,6 +95,8 @@ pub enum TypeSignature {
     VariadicEqual,
     /// One or more arguments with arbitrary types
     VariadicAny,
+    /// arbitrary number of arguments out of a possibly different but limited set of valid types
+    VariadicLimited(Vec<DataType>),
```

Comment:
Now we have `Variadic`, `VariadicLimited` and `VariadicEqual`, `VariadicAny`. Their behaviors are similar, and I'm wondering how we can better reflect their difference in naming.

The tail `Limited` also applies to `Variadic`. So `Variadic` is in actual "variadic" + "equal" + "limited", and the new one is "variadic" + "any" + "limited". How about changing `Variadic` to `VariadicLimitedEqual` and naming the new one `VariadicLimitedAny`? But they are a bit verbose, is there any other options?

Reply:
I had similar thoughts about the ambiguity of the names and went with `VariadicLimited` because I didn't want to change too much.

I am pro-verbosity: tab-completion makes verbosity less painful but there is no similar thing for ambiguity.